File: src/client/java/net/greenjab/fixedminecraft/hud/HUDOverlayEvent.java
package net.greenjab.fixedminecraft.hud;

import net.minecraft.client.gui.GuiGraphicsExtractor;

/** Credit: Squeek502 */
public class HUDOverlayEvent
{
    public int x;
    public int y;
    public GuiGraphicsExtractor context;
    public boolean isCanceled = false;

    public static class Saturation extends HUDOverlayEvent {
        public final float saturationLevel;
        public Saturation(float saturationLevel, int x, int y, GuiGraphicsExtractor context) {
            super(x, y, context);
            this.saturationLevel = saturationLevel;
        }
    }

    private HUDOverlayEvent(int x, int y, GuiGraphicsExtractor context) {
        this.x = x;
        this.y = y;
        this.context = context;
    }

}

File: src/client/kotlin/net/greenjab/fixedminecraft/hud/HUDOverlayEvent.kt
package net.greenjab.fixedminecraft.hud

import net.minecraft.client.gui.DrawContext


open class HUDOverlayEvent
private constructor(var x: Int, var y: Int, var context: DrawContext) {

    class Exhaustion(val exhaustion: Float, x: Int, y: Int, context: DrawContext) : HUDOverlayEvent(x, y, context) {
    }

    class Saturation(val saturationLevel: Float, x: Int, y: Int, context: DrawContext) : HUDOverlayEvent(x, y, context) {
    }

    var isCanceled: Boolean = false
}
